src/hammer-vlsi/technology/asap7/__init__.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+#  asap7 plugin for Hammer.
+#
+#  See LICENSE for licence details.
+
+import re
+import os
+import tempfile
+import shutil
+
+import gdspy
+from hammer_tech import HammerTechnology
+
+class ASAP7Tech(HammerTechnology):
+    """
+    override the HammerTechnology used in `hammeer_tech.py`
+    this class is loaded by function `load_from_json`, and will pass the `try` in `importlib`.
+    """
+    def post_install_script(self) -> None:
+        self.remove_duplication_in_drc_lvs()
+        self.generate_multi_vt_gds()
+        self.fix_sram_cdl_bug()
+
+    def remove_duplication_in_drc_lvs(self) -> None:
+        """
+        fix the conflicting in vendor drc/lvs deck between hammer-mentor-plugin.
+        """
+        self.logger.info("remove LAYOUT PATH|LAYOUT PRIMARY|LAYOUT SYSTEM|DRC RESULTS DATABASE|DRC SUMMARY REPORT|LVS REPORT|LVS POWER NAME|LVS GROUND NAME in DRC/LVS Decks")
+        ruledirs = os.path.join(self.extracted_tarballs_dir, "ASAP7_PDKandLIB.tar/ASAP7_PDKandLIB_v1p5/asap7PDK_r1p5.tar.bz2/asap7PDK_r1p5/calibre/ruledirs")
+        drc_deck = os.path.join(ruledirs, "drc/drcRules_calibre_asap7_171111a.rul")
+        lvs_deck = os.path.join(ruledirs, "lvs/lvsRules_calibre_asap7_160819a.rul")
+        pattern = re.compile(".*(LAYOUT\ PATH|LAYOUT\ PRIMARY|LAYOUT\ SYSTEM|DRC\ RESULTS\ DATABASE|DRC\ SUMMARY\ REPORT|LVS\ REPORT|LVS\ POWER NAME|LVS\ GROUND\ NAME).*\n")
+        with tempfile.NamedTemporaryFile(delete=False) as tf:
+            with open(drc_deck, 'r') as f:
+                tf.write(pattern.sub("", f.read()).encode('utf-8'))
+            shutil.copystat(drc_deck, tf.name)
+            shutil.copy(tf.name, drc_deck)
+
+        with tempfile.NamedTemporaryFile(delete=False) as tf:
+            with open(lvs_deck, 'r') as f:
+                tf.write(pattern.sub("", f.read()).encode('utf-8'))
+            shutil.copystat(lvs_deck, tf.name)
+            shutil.copy(tf.name, lvs_deck)
+
+    def generate_multi_vt_gds(self) -> None:
+        """
+        vendor only provide SLVT gds, this patch will generate other 3(LVT, RVT, SRAM) VT gds file.
+        """
+        self.logger.info("generate gds for Multi-VT cells")
+
+        orig_gds = os.path.join(self.extracted_tarballs_dir, "ASAP7_PDKandLIB.tar/ASAP7_PDKandLIB_v1p5/asap7libs_24.tar.bz2/asap7libs_24/gds/asap7sc7p5t_24.gds")
+        # load original_gds
+        asap7_original_gds = gdspy.GdsLibrary(infile=orig_gds)
+        original_cells = asap7_original_gds.cell_dict
+        # WTF is this?
```

Comment:
I don't understand if this comment is about what that dictionary entry is or about why its being deleted.